app/index.js

```diff
@@ -381,6 +383,26 @@ app.on('ready', () => {
       app.quit()
     })
 
+    ipcMain.on(messages.PREFS_RESTART, () => {
+      var message = locale.translation('prefsRestart')
+
+      appActions.showMessageBox({
+        buttons: [locale.translation('yes'), locale.translation('no')],
+        options: {
+          persist: false
+        },
+        message
+      })
+      prefsRestartCallbacks[message] = (buttonIndex, persist) => {
+        if (buttonIndex === 0) {
```

Comment:
I think this is sometimes being called before the pref is actually saved. if i have flash enabled and disable in prefs then click 'yes' to restart, it is still enabled when Brave relaunches.